### PR TITLE
feat: DataTransferManager, DataPackagePropertySet

### LINKED ISSUE #4716: Add support for DataPackageView.Properties
## What would you like to be added:

`DataPackageView.Properties`

## Why is this needed:

`TabView` samples use it

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
  - [x] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### LINKED ISSUE #4800: Add support for DataPackage.Properties
## What would you like to be added:

`DataPackage.Properties` 

## Why is this needed:

Drag & drop, sharing

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
  - [x] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### LINKED ISSUE #4801: Support for DataPackageManager
## What would you like to be added:

Sharing via `DataPackageManager`

## Why is this needed:

(☞ﾟヮﾟ)☞

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [x] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #4800, closes #4716, closes #4801

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Not supported

## What is the new behavior?

Support for:

- `DataPackagePropertySet`
- `DataPackagePropertySetView`
- `DataTransferManager`

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Application.cs [hunk 1]
```diff
@@ -104,6 +104,8 @@ public ApplicationTheme RequestedTheme
 			set
 			{
 				_requestedTheme = value;
+				// Sync with core application's theme
+				CoreApplication.RequestedTheme = value == ApplicationTheme.Dark ? SystemTheme.Dark : SystemTheme.Light;
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Application.cs:108]: How is this related to DateTransferManager?

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Application.cs:108]: It is used on iOS to support theme selection on `ShareUIOptions`. As it is in Uno.UWP, I can't directly access `Application.RequestedTheme` there, so I needed to replicate that on `CoreApplication`

File: src/Uno.UWP/ApplicationModel/DataTransfer/DataTransferManager.macOS.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+			if (dataPackageView.Contains(StandardDataFormats.Text))
+			{
+				var text = await dataPackageView.GetTextAsync();
+				sharedData.Add(new NSString(text));
+			}
+
+			var uri = await GetSharedUriAsync(dataPackageView);
+			if (uri != null)
+			{
+				sharedData.Add(NSUrl.FromString(uri.ToString()));
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/DataTransfer/DataTransferManager.macOS.cs:44]:
```suggestion
				sharedData.Add(NSUrl.FromString(uri.OriginalString));
```

File: src/Uno.UWP/ApplicationModel/DataTransfer/DataRequest.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		public DataPackage Data { get; set; } = new DataPackage();
+
+		public DateTimeOffset Deadline { get; }
+
+		public void FailWithDisplayText(string value)
+		{
+			throw new NotImplementedException();
+		}
+
+		public DataRequestDeferral GetDeferral()
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UWP/ApplicationModel/DataTransfer/DataRequest.cs:27]: Implemented instead 😀

**Review by jeromelaban — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]
